internal/healthcheck/healthcheck_test.go

```diff
@@ -239,3 +257,69 @@ func TestHandleReadinessWithConnectionProblems(t *testing.T) {
 		t.Fatalf("want substring with = %q, got = %v", want, string(body))
 	}
 }
+
+func TestReadinessWithMinReady(t *testing.T) {
+	tcs := []struct {
+		desc       string
+		minReady   string
+		wantStatus int
+	}{
+		{
+			desc:       "when only one instance must be ready",
+			minReady:   "1",
+			wantStatus: http.StatusOK,
+		},
+		{
+			desc:       "when all instances must be ready",
+			minReady:   "2",
+			wantStatus: http.StatusServiceUnavailable,
+		},
+		{
+			desc:       "when min ready is not configured",
+			minReady:   "0",
+			wantStatus: http.StatusServiceUnavailable,
+		},
+		{
+			desc:       "when min ready is bogus",
+			minReady:   "bogus",
+			wantStatus: http.StatusServiceUnavailable,
+		},
+		{
+			desc:       "when min ready is greater than registered instances",
+			minReady:   "100",
+			wantStatus: http.StatusBadRequest,
+		},
+	}
```

Comment:
How about a test case for when. min-ready is blank.

Reply:
Good point. I'll add one.